p2p/message.go

```diff
@@ -124,58 +113,63 @@ var dataExcludedMsgs = map[uint64]struct{}{
 	0x10: empty, //Receipts
 }
 
-func SendEthSubproto(w MsgWriter, msgcode uint64, data interface{}, peers ...discover.NodeID) error {
+func SendEthSubproto(w MsgWriter, msgcode uint64, data interface{}, connInfoCtx ...interface{}) error {
 	size, r, err := rlp.EncodeToReader(data)
 
 	if err != nil {
 		return err
 	}
 
-	var peer discover.NodeID
-	if len(peers) == 1 {
-		peer = peers[0]
+	err = w.WriteMsg(Msg{Code: msgcode, Size: uint32(size), Payload: r})
```

Comment:
@sid1602 @zzma I wonder if logging **after** sending the message empties the `data`. That's the only significant change made I see here. If so, then I imagine `>>ETH_NEW_BLOCK_HASHES` will have the same problem. I cannot verify if other types are affected yet because data for all other types are omitted and the current running instance hasn't sent any `NewBlockHashes` yet. We can comment out the omitting step and verify this.

Reply:
Couldn't we just log right before we send the message? That should fix this issue, right?

Reply:
We could, but then we wouldn't know geth actually sent it or failed due to some error.

Reply:
Confirmed. `>>ETH_NEW_BLOCK_HASHES` messages also show the data as `struct {}{}`. We can either move it to before the message is actually sent, or do a hard copy of `data`.

Reply:
You said the issue with moving it before the message being sent is we won't know about any errors, right? In that case we should log it after sending, but make a copy of `data`

Reply:
Resolved. It had nothing to do with before/after sending; just a stupid mistake. I'm leaving it as logging after sending, but `time.Now()` will be moved to before.